spanner/client_test.go

```diff
@@ -208,6 +208,74 @@ func TestClient_Single_Read_SessionNotFound(t *testing.T) {
 	}
 }
 
+func TestClient_Single_WhenInactiveTransactionsAndSessionIsNotFoundOnBackend_RemoveSessionFromPool(t *testing.T) {
+	t.Parallel()
+	server, client, teardown := setupMockedTestServerWithConfig(t, ClientConfig{
+		SessionPoolConfig: SessionPoolConfig{
+			MinOpened:                 1,
+			MaxOpened:                 1,
+			healthCheckSampleInterval: 1 * time.Second, // maintainer runs every 1s
+			InactiveTransactionRemovalOptions: InactiveTransactionRemovalOptions{
+				CloseInactiveTransactions: true,
+				executionFrequency:        1 * time.Second, // check long-running sessions every 1s
+			},
+		},
+	})
+	defer teardown()
+	server.TestSpanner.PutExecutionTime(
+		MethodExecuteStreamingSql,
+		SimulatedExecutionTime{Errors: []error{newSessionNotFoundError("projects/p/instances/i/databases/d/sessions/s")}},
+	)
+	ctx := context.Background()
+	single := client.Single()
+	iter := single.Query(ctx, NewStatement(SelectSingerIDAlbumIDAlbumTitleFromAlbums))
+	p := client.idleSessions
+	sh := single.sh
+	// simulate session to be checked out for more than 60mins
+	sh.mu.Lock()
+	sh.checkoutTime = time.Now().Add(-time.Hour)
+	sh.mu.Unlock()
+	// Allow maintainer to clean up long-running sessions
+	time.Sleep(3 * time.Second)
```

Comment:
Speaking out loud,
1. We cannot manually trigger this [cleanup function](https://github.com/googleapis/google-cloud-go/pull/8177/files#diff-9479303281d57658abd89eb39a8b1537fb58fca056d655901309ed1839c443f3R1482) as it is embedded with in the existing [maintainer()](https://github.com/googleapis/google-cloud-go/blob/2b99e4f39be20fe21e8bc5c1ec1c0e758222c46e/spanner/session.go#L1344), which is a background task. 
2. As the cleanup functionality is with in the background task we need to wait until it gets triggered next time.

I feel we cannot avoid sleep, but we can reduce the sleep time to `milliseconds`.  This way the unit tests do not take much longer time.

Reply:
@olavloite 
I have moved the cleanup logic to a separate function and invoking this method at regular intervals using the existing maintainer (background task). This way I need not rely on invoking the maintainer in unit tests and instead manually trigger the cleanup function to clean unexpectedly long-running sessions.

Also this eliminates the need to use `sleep` in unit tests.
However, I have one unit test to test if the maintainer is successful in invoking the clean up task and this would need `sleep` (in order of `milliseconds`).

I hope this is good now. Please let me know if something needs to be changed here.